server/src/main/java/org/opensearch/common/settings/Setting.java

```diff
@@ -1250,6 +1250,39 @@ private static boolean isFiltered(Property[] properties) {
         return properties != null && Arrays.asList(properties).contains(Property.Filtered);
     }
 
+    /**
+     * A writeable validator able to check the value of string type custom setting by using regular expression
+     */
+    public static class RegexValidator implements Writeable, Validator<String> {
+        private Pattern pattern;
+
+        public RegexValidator(String regex) {
```

Comment:
Add a javadoc for this constructor to tell users what the string means.  "A regular expression containing the only valid input for this setting."

Reply:
You added the javadoc to the streaminput constructor below, not this one... this is the one developers will use.  The StreamInput constructor will be used by the transport service sending these between OpenSearch and SDK.

Also you need the `@param` javadoc tag for the parameters.